### PR TITLE
Polynomial initialization of geodesics in DirichletDistributions

### PR DESCRIPTION
**Pull request by alebrigant**
- Add `_approx_geodesic_bvp` method that solves an approximation of the boundary value problem by minimizing the energy among curves whose coefficients are polynomial functions of time
- Use this in `_geodesic_bvp` as an initialization when solving the geodesic ODE.

### CODE REVIEW

File: geomstats/information_geometry/dirichlet.py [hunk 1]
```diff
@@ -480,12 +483,143 @@ def exp(self, tangent_vec, base_point, n_steps=N_STEPS):
...
+        """
+
+        def cost_fun(param):
+            """Compute the energy of the polynomial curve defined by param.
+
+            Parameters
+            ----------
+            param : array-like, shape=(degree - 1, dim)
+                Parameters of the curve coordinates' polynomial functions of time.
+            """
```
Comment (ninamiolane) [inline geomstats/information_geometry/dirichlet.py:540]: Missing Returns

File: geomstats/information_geometry/dirichlet.py [hunk 2]
```diff
@@ -480,12 +483,143 @@ def exp(self, tangent_vec, base_point, n_steps=N_STEPS):
...
+            energy = gs.sum(velocity_sqnorm) / n_times
+            return energy, length, curve, velocity
+
+        def cost_jacobian(param):
+            """Compute the jacobian of the cost function at polynomial curve defined by param.
+
+            Parameters
+            ----------
+            param : array-like, shape=(degree - 1, dim)
+                Parameters of the curve coordinates' polynomial functions of time.
```
Comment (ninamiolane) [inline geomstats/information_geometry/dirichlet.py:567]: Missing Returns

File: geomstats/information_geometry/dirichlet.py [hunk 3]
```diff
@@ -480,12 +483,143 @@ def exp(self, tangent_vec, base_point, n_steps=N_STEPS):
...
         initial_point,
         end_point,
-        n_steps=N_STEPS,
-        jacobian=False,
+        degree=5,
+        method="BFGS",
+        n_times=200,
+        jac_on=True,
+    ):
+        """Solve approximation of the geodesic boundary value problem.
```
Comment (ninamiolane) [inline geomstats/information_geometry/dirichlet.py:495]: Do you have a reference for this?

**Review by ninamiolane — APPROVED** [review]
LGTM once deep source passes and the docstrings comments are addressed!

File: geomstats/information_geometry/dirichlet.py (hunk 3 above)
Comment (alebrigant) [reply geomstats/information_geometry/dirichlet.py:495]: No not yet, but hopefully I will have one soon.

**Pull request merged**